www/class/centreonAuth.SSO.class.php

```diff
@@ -103,7 +102,17 @@ public function __construct(
             # Get configured values
             $clientId = $this->ssoOptions['openid_connect_client_id'];
             $clientSecret = $this->ssoOptions['openid_connect_client_secret'];
-            $redirectNoEncode = $this->ssoOptions['openid_connect_redirect_url'];
+            if (empty($this->ssoOptions['openid_connect_redirect_url'])) {
+                $redirectNoEncode = '{$scheme}://{$hostname}:{$port}/centreon/index.php';
```

Comment:
$redirectNoEncode = '{$scheme}://{$hostname}:{$port}' . rtrim($centreon->optGen["oreon_web_path"], "/") . '/index.php';

Reply:
Oreon - on the trail of history

I was already looking for something like that. Your hint is good but had another challenge: $centreon is not available during this state of authentication. The Centreon constructor always requires a userinfo to create an object. It took me a while until I realized I have to query the database for it. Like it is done in centreonLog.class.php. Maybe another thing that should be refactored one day. But not in this pull request.